src/profiler.cpp

```diff
@@ -783,6 +789,21 @@ void Profiler::writeLog(LogLevel level, const char* message, size_t len) {
     _jfr.recordLog(level, message, len);
 }
 
+void Profiler::setTraceContext(const char* trace_id, const char* span_id) {
+    free(current_trace_id);
+    free(current_span_id);
+    
+    current_trace_id = trace_id ? strdup(trace_id) : nullptr;
```

Comment:
According to the [spec](https://www.w3.org/TR/trace-context/#trace-id), trace-id is 16 byte and span-id is 8 byte. Thus, we don't need to `strdup`: declaring the fields as `thread_local char trace_id[16]` and copying them with `memcpy` will be faster.